### PR TITLE
Add feature flag proxy configurations with external secret support

### PR DESCRIPTION
**Pull request by arya-harness**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -49,8 +49,7 @@ RUN \
 # Cleanup
 RUN \
   apt-get clean && \
-  rm -fr /var/lib/apt/lists/* && \
-  rm -fr /tmp/*
```
Comment (arya-harness) [inline Dockerfile:53]: @Vbhatia20 can answer why this needed to deleted.. but without this it was not working.

File: charts/infra/ff-proxy/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "v2.0.0-rc.13"
```
Comment (jcox250) [inline charts/infra/ff-proxy/Chart.yaml]: Should we use the newest app version `2.0.3`?

**Review by jcox250 — COMMENTED** [review]
I think this looks good overall. It makes sense to place it inside the `charts` directory and I understand the intention behind hiding it in `charts/infra`. However, I'm a bit concerned about the possibility of customers finding this Helm chart instead of the one in `charts/ff-proxy`, deploying their proxy with it, and then requesting changes that we can't accommodate due to our internal requirements. If they're using `charts/ff-proxy`, we can make changes without affecting our internal infrastructure.

Are there additional steps we can take to clarify that this chart is intended for internal use only, and that `charts/ff-proxy` is the recommended option for customers? A couple of suggestions I can think of are

- Place the chart inside a `charts/internal/` directory and explicitly label it as internal use only, with a note redirecting customers to `charts/ff-proxy`.
- Move it to an `/infra/charts` directory. I think this is where a customer would be least likely to stumble across it and they do find it we can still redirect them to `charts/ff-proxy`.

Let me know your thoughts or if you have any other ideas.

File: charts/infra/ff-proxy/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# ff-proxy Helm Chart
```
Comment (jcox250) [inline charts/infra/ff-proxy/README.md]: This `README` looks like it's just a copy of the one from `charts/ff-proxy`. Should we write our own README here for this chart and explicitly call out it's for internal use and redirect any customers that find it to `charts/ff-proxy`

File: charts/infra/ff-proxy/Chart.yaml (hunk 2 above)
Comment (arya-harness) [reply charts/infra/ff-proxy/Chart.yaml]: its internal and gets autopopulated by pipelines while packing, so it should not matter.

File: charts/infra/ff-proxy/README.md (hunk 3 above)
Comment (arya-harness) [reply charts/infra/ff-proxy/README.md]: Agreed

**Review by jcox250 — APPROVED** [review]

**Pull request merged**